src/state/slices/backdrop_slice.ts

```diff
@@ -27,13 +27,35 @@ type BackdropSliceActions = {
    * the current backdrop key is `null`.
    */
   setBackdropVisible: (visible: boolean) => void;
+  /**
+   * Sets the brightness of the backdrop image, clamped to a percentage in the
+   * `[0, 200]` range.
+   * @throws {Error} If the brightness is `NaN`.
+   */
   setBackdropBrightness: (brightness: number) => void;
+  /**
+   * Sets the saturation of the backdrop image, clamped to a percentage in the `[0,
+   * 100]` range.
+   * @throws {Error} If the saturation is `NaN`.
+   */
   setBackdropSaturation: (saturation: number) => void;
+  /**
+   * Sets the opacity of objects when the backdrop is visible, clamped to a
+   * percentage in the `[0, 100]` range.
+   * @throws {Error} If the opacity is `NaN`.
+   */
   setObjectOpacity: (opacity: number) => void;
 };
 
 export type BackdropSlice = BackdropSliceState & BackdropSliceActions;
 
+const clampPercentage = (value: number, min: number, max: number): number => {
```

Comment:
maybe name this `clampWithNaNCheck` and move it into a shareable utility module?